### PR TITLE
Add get methods for ListCommand predicates, for category and timespan.

### LINKED ISSUE #134: Filter by category/time function performing "or" combination of predicates
When 2 filters are applied (category and timespan), the resulting list contains elements which fulfill either one of the predicates.
The result list should contain elements which fulfill both predicates instead.

### PR DESCRIPTION
**Pull request by jinbesan**
Fix bug that list would still list all expenses in a timespan regardless of category even if both tags present

Closes #134

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -207,20 +216,40 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
...
+            return Timespan.WEEK;
         }
         if (trimmedTimespan.equals("month") || trimmedTimespan.equals("m")) {
-            return now.withDayOfMonth(1);
+            return Timespan.MONTH;
         }
         if (trimmedTimespan.equals("year") || trimmedTimespan.equals("y")) {
-            return now.withDayOfYear(1);
+            return Timespan.YEAR;
         }
```
Comment (shirsho-12) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:232]: Good use of enums

**Review by shirsho-12 — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/logic/parser/ListCommandParserTest.java [hunk 2]
```diff
@@ -33,7 +33,7 @@ public void parse_categoryFieldPresent_success() {
 
     @Test
     public void parse_timespanFieldPresent_success() {
-        ExpenseInTimespanPredicate predicate = new ExpenseInTimespanPredicate(LocalDate.now().withDayOfYear(1));
+        ExpenseInTimespanPredicate predicate = new ExpenseInTimespanPredicate(YEAR);
```
Comment (gitsac) [inline src/test/java/seedu/address/logic/parser/ListCommandParserTest.java:36]: Good to have abstracted the previous argument out to use an enum instead!

**Review by gitsac — APPROVED** [review]
LGTM!

**Pull request merged**